test/integration/ratelimit_integration_test.cc

```diff
@@ -172,6 +172,7 @@ TEST_P(RatelimitIntegrationTest, Timeout) {
   switch (clientType()) {
   case Grpc::ClientType::EnvoyGrpc:
     test_server_->waitForCounterGe("cluster.ratelimit.upstream_rq_timeout", 1);
+    test_server_->waitForCounterGe("cluster.ratelimit.upstream_rq_504", 1);
```

Comment:
Just curious, why do we not need to wait on `streams_total` to be incremented below, too?

Reply:
The Google gRPC client increments `streams_total` when .streams come to life and then increments `streams_total_<grpc status code>` when they close. So, it's sufficient to wait on the close. For Envoy gRPC, it appears that `rq_timeout` is bumped before `rq_504`. Could have just waiting on `rq_504` I suppose, but belts-and-braces.

Reply:
Gotcha.  Makes sense.  It probably makes sense to not depend on implementation defined order of modification for `timeout` and `504` anyway.